ui/StatusQ/include/StatusQ/aggregator.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <QAbstractListModel>
+#include <QObject>
+#include <QVariant>
+
+class Aggregator : public QObject {
+    Q_OBJECT
+
+    Q_PROPERTY(QAbstractItemModel* model READ model WRITE setModel NOTIFY modelChanged)
+    Q_PROPERTY(QString roleName READ roleName WRITE setRoleName NOTIFY roleNameChanged)
+    Q_PROPERTY(QVariant value READ value NOTIFY valueChanged)
```

Comment:
If `value` is always a number, then return `double` directly here, not QVariant

Reply:
I used `QVariant` here bc it's an abstract class and we can potentially want another kind of aggregation, for example.. string aggregation / concatenation for a given `roleName`.  Then, this is more generic..